src/shared/bus-unit-util.c

```diff
@@ -410,6 +410,51 @@ static int bus_append_exec_command(sd_bus_message *m, const char *field, const c
         return 1;
 }
 
+static int bus_append_open_file(sd_bus_message *m, const char *field, const char *eq) {
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        int r;
+
+        assert(m);
+
+        r = open_file_parse(eq, &of);
+        if (r < 0)
+                return bus_log_create_error(r);
```

Comment:
This needs a custom error message instead of `bus_log_create_error()`. That function is only valid when creating D-Bus objects which is not what we're doing here.

You something like: `Failed to parse OpenFile= setting: %m`